agent/lib/src/health.dart

```diff
@@ -142,3 +144,42 @@ Future<HealthCheckResult> removeCachedData(
   }
   return HealthCheckResult.success();
 }
+
+/// Closes system dialogs on iOS, e.g. the one about new system update.
+///
+/// The dialogs often cause test flakiness and performance regressions.
+@visibleForTesting
+Future<HealthCheckResult> closeIosDialog(
+    {ProcessManager pm = const LocalProcessManager(), DeviceDiscovery discovery, Directory dialogDir}) async {
+  if (discovery == null) {
+    discovery = devices;
+  }
+  List<Device> ds = await discovery.discoverDevices();
+  for (Device d in ds) {
+    if (!(d is IosDevice)) {
+      continue;
+    }
```

Comment:
How about using `Device d = ds.where((Device d) => d is IosDevice)` to remove the for loop?